src/main/java/dev/emi/trinkets/api/client/TrinketRenderer.java

```diff
@@ -1,19 +1,60 @@
 package dev.emi.trinkets.api.client;
 
 import dev.emi.trinkets.api.Trinket;
+import net.minecraft.client.MinecraftClient;
 import net.minecraft.client.network.AbstractClientPlayerEntity;
 import net.minecraft.client.render.VertexConsumerProvider;
+import net.minecraft.client.render.entity.EntityRenderer;
+import net.minecraft.client.render.entity.LivingEntityRenderer;
+import net.minecraft.client.render.entity.model.BipedEntityModel;
+import net.minecraft.client.render.entity.model.EntityModel;
 import net.minecraft.client.render.entity.model.PlayerEntityModel;
 import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.entity.LivingEntity;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.math.Vec3f;
 
 public interface TrinketRenderer {
 	float MAGIC_ROTATION = 180f / (float) Math.PI;
 
-	void render(ItemStack stack, Trinket.SlotReference slot, MatrixStack matrices, VertexConsumerProvider vertexConsumer, int light,
-			PlayerEntityModel<AbstractClientPlayerEntity> model, AbstractClientPlayerEntity player, float limbAngle, float limbDistance,
-			float tickDelta, float animationProgress, float headYaw, float headPitch);
+	/**
+	 * Renders the Trinket
+	 *
+	 * @param stack The {@link ItemStack} for the Trinket being rendered
+	 * @param slotReference The exact slot for the item being rendered
+	 * @param contextModel The model this Trinket is being rendered on
+	 */
+	void render(ItemStack stack, Trinket.SlotReference slotReference, EntityModel<? extends LivingEntity> contextModel,
+				MatrixStack matrices, VertexConsumerProvider vertexConsumers, int light, LivingEntity entity,
+				float limbAngle, float limbDistance, float tickDelta, float animationProgress, float headYaw,
+				float headPitch);
+
+	/**
+	 * Rotates the rendering for the models based on the entity's poses and movements. This will do
+	 * nothing if the entity render object does not implement {@link LivingEntityRenderer} or if the
+	 * model does not implement {@link BipedEntityModel}).
+	 *
+	 * @param entity The wearer of the trinket
+	 * @param model The model to align to the body movement
+	 */
+	static void followBodyRotations(final LivingEntity entity,
+									final BipedEntityModel<LivingEntity> model) {
```

Comment:
Is there a particular reason that this was modified from Curios to take only one model?

Reply:
Correct me if I'm wrong but when you create a BipedModel this way for your trinket I don't expect anyone to supply more than one model, ever. I just don't really see any use for it and maybe makes the api a bit less confusing and it's implementation a bit more readable.

Reply:
That's fair. I was just mainly curious if there was a more major reason for it, I think that's fine as it is. It's just a helper method anyway, so it doesn't necessarily need to cover every use-case.